Stylefixes

dokato opened the pull request:
This addresses #212 - makes code coherent with new STYLEGUIDE.

**DoD**

- [x] Major project work has a corresponding task. If there’s no task for what you are doing, create it. Each task needs to be well defined and described.

- [x] Change has been tested (manually or with automated tests), everything runs correctly and works as expected. No existing functionality is broken.

- [x] No new error or warning messages are introduced.

- [x] All interaction with semantic functions, examples and docs are written from the perspective of the person using or receiving it. They are understandable and helpful to this person.

- [x] If the change affects code or repo structure, README, documentation and code comments should be updated. 

- [x] All code has been peer-reviewed before merging into any main branch.

- [x] All changes have been merged into the main branch we use for development (develop).

- [x] Continuous integration checks (linter, unit tests) are configured and passed.

- [x] Unit tests added for all new or changed logic.

- [x] All task requirements satisfied. The reviewer is responsible to verify each aspect of the task.

- [x] Any added or touched code follows our style-guide.

olgamie commented on R/date_input.R:
```diff
@@ -43,9 +43,10 @@
 #' @export
-date_input <- function(name, label = NULL, value = NULL, min = NULL, max = NULL,
+#' @rdname date_input
+date_input <- function(input_id, label = NULL, value = NULL, min = NULL, max = NULL,
```
Is it possible to change the function, so it takes a character as an icon e.g. "dog" and do uiicon("dog") inside the function? We could then call shiny.semantic:::uiicon(), and don't export uiicon. Then the user would only have icon function exported.

olgamie commented on R/dsl.R:
```diff
@@ -361,10 +356,10 @@ uimessage <- function(header, content, class = "", icon, closable = FALSE) {
 #'       title = "My page",
-#'       uimenu(menu_item("Menu"),
+#'       menu(menu_item("Menu"),
 #'              uidropdown(
```
please fix examples, old uidropdown and uiicon

olgamie commented on R/dsl.R:
```diff
@@ -430,10 +425,10 @@ menu_item <- function(..., item_feature = "", style = NULL, href = NULL) {
 #' @examples
 #'
-#' uidropdown(
+#' dropdown_menu(
```
Please fix examples

dokato replied:
here again not sure what you're referring too, this example worked for me

olgamie commented on R/input.R:
```diff
@@ -25,13 +25,15 @@ uiinput <- function(..., class = "") {
+#' @param width The width of the input, eg. "40px"
 #'
 #' @details
 #' The following \code{type}s are allowed:
```
spaces are missing between item{}(space missing here){}

olgamie commented on R/input.R:
```diff
@@ -25,13 +25,15 @@ uiinput <- function(..., class = "") {
 #' @param type Change depending what type of input is wanted. See details for options.
 #' @param placeholder Text visible in the input when nothing is inputted.
 #' @param attribs A named list of attributes to assign to the input.
+#' @param label character with label put above the input
```
The label is put on the left side.

olgamie commented on R/input.R:
```diff
@@ -52,33 +54,87 @@ uiinput <- function(..., class = "") {
-#'   tags$label("Text input"),
-#'   uitextinput("ex", type = "text", placeholder = "Enter Text")
-#' )
+#' text_input("ex", label = "Your text", type = "text", placeholder = "Enter Text")
```
Please change to
```
library(shiny)
library(shiny.semenitc)
if (interactive()) {
   ui <- semanticPage(
      uiinput(
         text_input("ex", label = "Your text", type = "text", placeholder = "Enter Text")
      )
   )
   server <- function(input, output, session) {
   }
   shinyApp(ui, server)
}
```

olgamie commented on R/input.R:
```diff
@@ -99,15 +155,15 @@ uitextinput <- function(name, value = "", type = "text", placeholder = NULL, att
 #'
 #' @export
-uinumericinput <- function(name, value, min = NA, max = NA, step = NA,
+numeric_input <- function(input_id, value, min = NA, max = NA, step = NA,
```
`text_input` has a label argument. Should we make this consistent and add lebel here as well, or remove from text?

olgamie commented on R/rating.R:
```diff
@@ -48,7 +62,7 @@ rating <- function(name, label = "", value = 0, max = 3, icon = "star",
 #' Update rating
 #'
-#' check \code{rating} to learn more.
+#' check \code{rating_input} to learn more.
```
This isn't very informative :D sth like: `Change the value of a rating input on the client` ?

olgamie commented on R/slider.R:
```diff
@@ -62,24 +62,46 @@
+
 #' Update slider Semantic UI component
 #'
 #' Change the value of a \code{\link{uislider}} input on the client.
```
Uislider => slider_input

olgamie commented on R/slider.R:
```diff
@@ -88,20 +110,33 @@ uirange <- function(name, value, value2, min, max, step = 1, class = NULL) {
 #' @param value The value to be selected for the sldier (lower value if using range).
 #'
-#' @seealso uislider
+#' @seealso slider
```
slider_input?

olgamie commented on R/utils.R:
```diff
@@ -23,3 +23,24 @@ check_proper_color <- function(color) {
   }
 }
+
+#' Generates random id of element
```
Can we provide more details when to use it and example?

dokato replied:
This is hidden function, not to see by the user, actually examples of hidden function cause problem when doing checks for CRAN, but I elaborated in the description.

olgamie commented on R/utils.R:
```diff
@@ -23,3 +23,24 @@ check_proper_color <- function(color) {
+
+#' Check for extra arguments
+#'
+#' This throws warning if there're parameters not supported by semantic.
```
Same here can we add example

dokato replied:
same as above

dokato commented on R/checkbox.R:
```diff
@@ -36,20 +36,20 @@
 #' @export
 toggle <- function(id, label = "", is_marked = TRUE, style = NULL) {
-  simple_checkbox(id, label, type = "toggle", is_marked = is_marked, style = style)
+  checkbox_input(id, label, type = "toggle", is_marked = is_marked, style = style)
```
hmm, here I don't understand what's wrong, for me works as charm, also we don't really use `shinyjs` here

olgamie replied:
So I had to add shiny.semantic::toggle to get the example work. It seems that previous error was due to toggle function coming from shinyjs.

```
library(shiny)
library(shiny.semantic)

ui <- semanticPage(
   shinyUI(
      shiny.semantic::toggle("tog1", "My Label", TRUE)
   )
)
server <- function(input, output, session) {
   observeEvent(input$tog1, {
      print(input$tog1)
   })
}
shinyApp(ui, server)
```

dokato commented on R/date_input.R:
```diff
@@ -1,6 +1,6 @@
 #' Define simple date input with semantic ui styling
```
it should be changed to fomantic at some point anyway probalby

dokato commented:
@olgamie I tried to also fix examples from `examples/` folder. Hope I covered all.

olgamie commented on R/dsl.R:
```diff
@@ -17,40 +17,40 @@
 #'     semanticPage(
 #'       # Basic icon
-#'       uiicon("home"),
+#'       icon("home"),
```
Unfortunately there is still a problem with rendering icons properly:
![image](https://user-images.githubusercontent.com/4321346/88024147-8c3f8d80-cb32-11ea-9bb6-3ac2adb1b938.png)

dokato commented:
Thanks @olgamie for catching these bugs. Icons example seem to be working for me, look below. `app.R` indeed was broken, fixed now. Export added too.

I introduced also some fixes to dropdown and update action button and there were some minor mistakes in roxygen compilation.
 
<img width="190" alt="Screenshot 2020-07-21 at 12 28 50" src="https://user-images.githubusercontent.com/4547289/88050831-7c827200-cb4f-11ea-9427-edfc86157465.png">

dokato commented:
@olgamie done
